### PR TITLE
RA-1910, configure paper ID column in search patients table

### PR DESCRIPTION
**Pull request by cioan**

### CODE REVIEW

**Comment by cioan** [comment]
@mogoodrich , this is the UHM-5273 ticket. Thanks!

File: omod/src/main/java/org/openmrs/module/registrationapp/page/controller/FindPatientPageController.java [hunk 1]
```diff
@@ -98,15 +100,16 @@ public void controller(UiSessionContext uiSessionContext,
...
-            model.addAttribute("paperRecordIdentifierDefinitionAvailable", true);
-            d.addColumn("paperRecordIdentifier", paperRecordIdentifierDefinition, "", new PropertyConverter(String.class, "identifier"));
-        }
-        else {
-            model.addAttribute("paperRecordIdentifierDefinitionAvailable", false);
+        
+        String gpPaperRecordIdDef = Context.getAdministrationService().getGlobalProperty(RegistrationAppConstants.GP_PAPER_RECORD_IDENTIFIER_DEFINITION);
+        model.addAttribute("paperRecordIdentifierDefinitionAvailable", false);
+        if (StringUtils.isNotBlank(gpPaperRecordIdDef)) {
+            // add the paper record identifier, if the definition is available (provided by the paper record module)
```
Comment (mogoodrich) [inline omod/src/main/java/org/openmrs/module/registrationapp/page/controller/FindPatientPageController.java:107]: let's change this to "if a definition is available (the default is provided by  the paper record module)

File: omod/src/main/resources/config.xml [hunk 2]
```diff
@@ -138,6 +138,13 @@
 	</messages>
 	<!-- /Internationalization -->
 
+	<globalProperty>
+		<property>registrationapp.paperRecordIdentifierDefinition</property>
+		<description>
```
Comment (mogoodrich) [inline omod/src/main/resources/config.xml:143]: for backwards compatibility, the default value here should be the one that is currently used ""paperrecord.patientDataDefinition.paperRecordIdentifier"

File: omod/src/main/resources/config.xml [hunk 3]
```diff
@@ -138,6 +138,13 @@
...
+			Paper record identifier definition
```
Comment (mogoodrich) [inline omod/src/main/resources/config.xml:144]: how about: Report Definition to retrieve the paper record identifier for a patient; default comes from paper record module

**Review by mogoodrich — CHANGES_REQUESTED** [review]
Added a few review comments, thanks @cioan !

**Comment by mseaton** [comment]
@cioan I agree with @mogoodrich .  The reason we took this approach was for backwards compatibility - we want things to work exactly like they currently do if someone upgrades (i.e. the fact that a new global property exists should be transparent, and all of the definition names should default to what they were hard-coded to before).

**Review by mogoodrich — APPROVED** [review]
LGTM, thanks @cioan !

**Pull request merged**